src/util/prolog/prologResult.ts

```diff
@@ -0,0 +1,54 @@
+export class PrologResult {
+	private readonly results: string[][];
+	private readonly rawResults: string[];
+
+	constructor(rawResults: string[]) {
+		this.rawResults = rawResults;
+		this.results = [];
+	}
+
+	public getResults(): string[][] {
+		if (this.results.length >= 1 ) { return this.results; }
+
+		this.rawResults.forEach((e: any) => {
+			this.results.push(this.rawResultsToInnerArray(e));
+		});
+
+		return this.results;
+	}
+
+	public filterBooleanResults(): string[][] {
```

Comment:
Since this method filters for boolean results, why don't you return `boolean`s?

Reply:
The idea was to keep it consistent with all the other outputs, but thinking about it now it didn't make a lot of sense.